### PR TITLE
DAOS-6556 container: add max oid to container properties.

### PR DESCRIPTION
**Pull request by mchaarawi**
- This allows fetching the max oid that the container is tracking.
- The max oid property can only be changed at container creation for now.
  if set_prop is used to set the max_oid prop, a -DER_NO_PERM error is
  returned. Support for this will be added in the future.
- Add a simple test for this.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/container/cli.c [hunk 1]
```diff
@@ -1111,11 +1114,21 @@ dc_cont_set_prop(tse_task_t *task)
...
 	struct cont_set_prop_args	 arg;
+	int				 i;
 	int				 rc;
 
 	args = dc_task_get_args(task);
 	D_ASSERTF(args != NULL, "Task Argument OPC does not match DC OPC\n");
 
+	for (i = 0; i < args->prop->dpp_nr; i++) {
+		if (args->prop->dpp_entries[i].dpe_type ==
+		    DAOS_PROP_CO_MAX_OID) {
```
Comment (kjacque) [inline src/container/cli.c:1125]: I guess it doesn't save many lines, but you could use daos_prop_entry_get to make this check a little simpler

**Review by kjacque — APPROVED** [review]

File: src/utils/daos_hdlr.c [hunk 2]
```diff
@@ -1241,6 +1241,14 @@ cont_decode_props(daos_prop_t *props, daos_prop_t *prop_acl)
 			D_PRINT("<unknown value> ("DF_X64")\n", entry->dpe_val);
 	}
 
+	entry = daos_prop_entry_get(props, DAOS_PROP_CO_MAX_OID);
+	if (entry == NULL) {
+		fprintf(stderr, "max oid property not found\n");
+		rc = -DER_INVAL;
+	} else {
+		D_PRINT("max oid:\t\t"DF_U64"\n", entry->dpe_val);
```
Comment (johannlombardi) [inline src/utils/daos_hdlr.c:1249]: I would put "alloced" somewhere in the name.

**Review by johannlombardi — DISMISSED** [review]
Looks good to me. I am just unsure about the name, since max means to me something that you cannot go above.

**Review by kjacque — DISMISSED** [review]

**Comment by johannlombardi** [comment]
many conflicting files, i am afraid ...

**Comment by mchaarawi** [comment]
> many conflicting files, i am afraid ...

ugh, Liang beat me with his patch :-)

**Review by kjacque — APPROVED** [review]

**Comment by mchaarawi** [comment]
sigh... another PR landed that generates more conflicts. will need to rebase again.

File: src/container/srv_internal.h [hunk 3]
```diff
@@ -119,6 +119,7 @@ struct cont_iv_prop {
 	uint64_t	cip_csum_server_verify;
 	uint64_t	cip_dedup;
 	uint64_t	cip_dedup_size;
+	uint64_t	cip_max_oid;
```
Comment (johannlombardi) [inline src/container/srv_internal.h:122]: confusing that the name does not patch the property name

File: src/container/cli.c [hunk 4]
```diff
@@ -936,6 +936,9 @@ cont_query_bits(daos_prop_t *prop)
 		case DAOS_PROP_CO_DEDUP_THRESHOLD:
 			bits |= DAOS_CO_QUERY_PROP_DEDUP_THRESHOLD;
 			break;
+		case DAOS_PROP_CO_ALLOCED_OID:
+			bits |= DAOS_CO_QUERY_PROP_MAX_OID;
```
Comment (johannlombardi) [inline src/container/cli.c:940]: Prone to error ...

**Review by johannlombardi — APPROVED** [review]
User interface looks fine, so i am ok to land this.
That being said, the internal naming disconnection is a bit confusing now.

File: src/container/cli.c (hunk 4 above)
Comment (mchaarawi) [reply src/container/cli.c:940]: yea i think i missed a few places to change the name, i will push a follow on PR

**Pull request merged**